File: src/Java/myservicebus-abstractions/src/main/java/com/myservicebus/PublishContext.java
package com.myservicebus;

import com.myservicebus.tasks.CancellationToken;
import com.myservicebus.serialization.MessageIntent;

/**
 * Specialized context used for publish operations.
 */
public class PublishContext extends SendContext {
    public PublishContext(Object message) {
        super(message);
        setIntent(MessageIntent.PUBLISH);
    }

    public PublishContext(Object message, CancellationToken cancellationToken) {
        super(message, cancellationToken);
        setIntent(MessageIntent.PUBLISH);
    }
}

File: src/MyServiceBus/PublishContext.cs
using System;
using System.Threading;
using MyServiceBus.Serialization;

namespace MyServiceBus;

public class PublishContext : SendContext, IPublishContext
{
    public PublishContext(Type[] messageTypes, IMessageSerializer messageSerializer, CancellationToken cancellationToken = default)
        : base(messageTypes, messageSerializer, cancellationToken)
    {
    }
}

